geomesa-gt/geomesa-gt-partitioning/src/main/scala/org/locationtech/geomesa/gt/partition/postgis/dialect/filter/LiteralFunctionVisitor.scala

```diff
@@ -0,0 +1,38 @@
+/***********************************************************************
+ * Copyright (c) 2013-2022 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.gt.partition.postgis.dialect.filter
+
+import org.geotools.factory.CommonFactoryFinder
+import org.geotools.filter.FilterAttributeExtractor
+import org.geotools.filter.visitor.DuplicatingFilterVisitor
+import org.opengis.filter.Filter
+import org.opengis.filter.expression.Function
+
+/**
+ * Replaces 'constant' functions with their literal values
+ */
+class LiteralFunctionVisitor extends DuplicatingFilterVisitor {
```

Comment:
Hmm...  I wonder if there's already a visitor like this in GeoTools.

Also, I should ask for a unit test... but I'mma :shipit: !

Reply:
there was a similar block of code in the postgis dialect, but it only replaced the arguments to a function and not the function itself.